pyuvdata/uvcal/uvcal.py

```diff
@@ -238,36 +294,70 @@ def __init__(self):
-        )
-        self._time_range = uvp.UVParameter(
-            "time_range", description=desc, form=2, expected_type=float, required=False
-        )
-
-        desc = (
-            "Array of calibration solution times, center of integration, "
-            "shape (Ntimes), units Julian Date"
+            "Array of calibration solution times, center of integration, shape "
+            "(Ntimes), units Julian Date. Should only be set cal solutions were "
+            "calculated per integration (rather than over a range of integrations). "
+            "Only one of time_range and time_array should be set."
         )
         self._time_array = uvp.UVParameter(
             "time_array",
             description=desc,
             form=("Ntimes",),
             expected_type=float,
             tols=1e-3 / (60.0 * 60.0 * 24.0),
+            required=False,
         )
 
-        # standard angle tolerance: 1 mas in radians.
-        desc = "Array of lsts, center of integration, shape (Ntimes), units radians"
+        desc = (
+            "Time range (in JD) that cal solutions are valid for. This should be an "
+            "array with shape (Ntimes, 2) where the second axis gives the start_time "
+            "and end_time (in that order) in JD. Should only be set if the cal "
+            "solutions apply over a range of times. Only one of time_range and "
+            "time_array should be set."
+        )
+        self._time_range = uvp.UVParameter(
+            "time_range",
+            description=desc,
+            form=("Ntimes", 2),
+            expected_type=float,
+            required=False,
+        )
+
+        desc = (
+            "Array of lsts, center of integration, shape (Ntimes), units radians. "
+            "Should only be set cal solutions were calculated per integration (rather "
+            "than over a range of integrations). Only one of lst_range and lst_array "
+            "should be set."
+        )
         self._lst_array = uvp.UVParameter(
             "lst_array",
             description=desc,
             form=("Ntimes",),
             expected_type=float,
             tols=uvutils.RADIAN_TOL,
-            required=True,
+            required=False,
         )
 
+        desc = (
+            "LST range (in JD) that cal solutions are valid for. This should be an "
+            "array with shape (Ntimes, 2) where the second axis gives the start_lst "
+            "and end_lst (in that order) in radians. Should only be set if the cal "
+            "solutions apply over a range of times. Only one of lst_range and "
+            "lst_array should be set."
+        )
+        self._lst_range = uvp.UVParameter(
+            "lst_range",
+            description=desc,
+            form=("Ntimes", 2),
+            expected_type=float,
+            tols=uvutils.RADIAN_TOL,
+            required=False,
+        )
+
+        # TODO:
```

Comment:
Should we turn these into issues?

Reply:
Actually, I think those are cruft. The first one is done and I think we sorted it out reasonably without the other package.